packages/server/lib/controllers/v1/environment/patchEnvironment.ts

```diff
@@ -0,0 +1,74 @@
+import { z } from 'zod';
+import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+import { asyncWrapper } from '../../../utils/asyncWrapper.js';
+import type { DBEnvironment, PatchEnvironment } from '@nangohq/types';
+import { environmentService } from '@nangohq/shared';
+import { environmentToApi } from '../../../formatters/environment.js';
+
+const validationBody = z
+    .object({
+        callback_url: z.string().url().optional(),
+        hmac_key: z.string().min(0).max(1000).optional(),
+        hmac_enabled: z.boolean().optional(),
+        slack_notifications: z.boolean().optional(),
+        otlp_endpoint: z.string().url().or(z.literal('')).optional(),
+        otlp_headers: z
+            .array(z.object({ name: z.string().min(1).max(256), value: z.string().min(1).max(4000) }))
```

Comment:
> 4000
generous 😉

Reply:
ahah yes, I wasn't sure what people were putting there but we still need a limit :D